sdk/dotnet/Pulumi.Tests/Serialization/EnumConverterTests.cs

```diff
@@ -0,0 +1,214 @@
+// Copyright 2016-2020, Pulumi Corporation
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Threading.Tasks;
+using Pulumi.Serialization;
+using Xunit;
+
+namespace Pulumi.Tests.Serialization
+{
+    public class EnumConverterTests : ConverterTests
+    {
+        [EnumType]
+        public readonly struct ContainerColor : IEquatable<ContainerColor>
```

Comment:
Here are the requirements for a struct-based enum to be able to be serialized/deserialized. It must:

 - Be a value type (`struct`) decorated with `EnumTypeAttribute`
 - Have a constructor that takes a single parameter of the underlying type (the constructor can be `private`)
 - Have an explicit conversion operator that converts the enum type to the underlying type
 - Have an underlying type of `String` or `Double`

Implementing `IEquatable`, adding `==`/`!=` operators, and overriding `ToString` isn't required, but is recommended and is what our codegen will do.